File: java/Container With Most Water.py
class Solution:
    # @return an integer
    def maxArea(self, height):
        if height == None:
            return 0;
            
        left = 0;
        right = len(height) - 1;
        
        l = right - left;
        
        max = min(height[left],height[right]) * l;
        
        while left < right: 
            if height[left] < height[right]:
                left += 1;
            else:
                right -= 1;
                
            if max < min(height[left],height[right]) * (right-left):
                max = min(height[left],height[right]) * (right-left);
                
        return max;

File: cpp/digit_problem/Container With Most Water .cpp
class Solution {
public:
    
    int twomin(int a,int b)
    {
        return a > b ? b : a;
    }
    
    int maxArea(vector<int> &height) {
        int left = 0;
        int right = height.size() - 1;
        int iret = 0;
        
        while(left < right)
        {
            int tmp = twomin(height[left],height[right]);
            
            if(tmp * (right - left) > iret)
            {
                iret = tmp * (right - left);
            }
            
            if(height[left] < height[right])
                left += 1;
            else
                right -= 1;
        }
        
        return iret;
    }
};